### PR TITLE
[manager] Add support for request Inputs/outputs

### PR DESCRIPTION
**Pull request by kparichay**
Added support for requesting inputs and outputs
Also update implementation to use a common function for easier
management

Signed-off-by: Parichay Kapoor <pk.kapoor@samsung.com>

### CODE REVIEW

File: nntrainer/tensor/manager.h [hunk 1]
```diff
@@ -538,6 +542,33 @@ class Manager {
...
+
+  /**
+   * @brief     Create tensors with the given spec
+   *
+   * @param w   node Graph node to extract node identifiers/info
+   * @param w   create tensors list
+   * @param layer_objs_list list to store the created tensors
+   */
+  template <typename T>
+  static std::vector<T *> _requestTensors(
```
Comment (zhoonit) [inline nntrainer/tensor/manager.h:554]: Could you check this?

s/_requestTensors/requestTensors_/

Comment (zhoonit) [reply nntrainer/tensor/manager.h:554]: Could you also update the docs?

File: nntrainer/tensor/manager.h [hunk 2]
```diff
@@ -169,6 +169,7 @@ class Manager {
   /**
    * @brief     Create tensors with the given spec
    *
+   * @param w   node Graph node to extract node identifiers/info
```
Comment (zhoonit) [inline nntrainer/tensor/manager.h:172]: Looks like for some reason, @param w was used everywhere before lol. Could you bother to change this?

File: nntrainer/tensor/manager.h (hunk 1 above)
Comment (kparichay) [reply nntrainer/tensor/manager.h:554]: did you mean _requestTensors -> requestTensors?

File: nntrainer/tensor/manager.h (hunk 2 above)
Comment (kparichay) [reply nntrainer/tensor/manager.h:172]: sure, done.

File: nntrainer/tensor/manager.h (hunk 1 above)
Comment (zhoonit) [reply nntrainer/tensor/manager.h:554]: the github markdown messed up the comment 🤣  `s/_requestTensors/requestTensors_/`

**Review by zhoonit — APPROVED** [review]
LGTM

File: nntrainer/tensor/manager.h (hunk 1 above)
Comment (kparichay) [reply nntrainer/tensor/manager.h:562]: updated  `s/_requestTensors/requestTensors/`

File: nntrainer/tensor/manager.h [hunk 3]
```diff
@@ -408,8 +419,9 @@ class Manager {
   std::vector<std::vector<std::unique_ptr<Weight>>>
     weights_v2; /**< weights for the layers */
   std::vector<std::vector<std::unique_ptr<Var_Grad>>>
-    inouts_v2; /**<
-              inputs/outputs for the layers */
+    inputs_v2; /**< inputs for the layers */
+  std::vector<std::vector<std::unique_ptr<Var_Grad>>>
+    outputs_v2; /**< outputs for the layers */
```
Comment (jijoongmoon) [inline nntrainer/tensor/manager.h:424]: are we going to keep this _v2 later?

**Review by jijoongmoon — APPROVED** [review]
LGTM

File: nntrainer/tensor/manager.h (hunk 3 above)
Comment (kparichay) [reply nntrainer/tensor/manager.h:424]: the names can be updated, that's upto us.
I kept v2 as weights already exist, so created weights_v2.

**Review by jijoongmoon — APPROVED** [review]
LGTM

**Pull request merged**